vuepress/docs/next/tutorials/create/ms/add-access-controls.md

```diff
@@ -1,109 +1,161 @@
 
-If you choose to use Realm-assigned roles then the code above would need to change. In the backend, use the following annotations: `@Secured('ROLE_ADMIN)` or `@PreAuthorize(hasRole('ROLE_ADMIN'))`. In the frontend, use `keycloak.hasRealmRole` instead of `keycloak.hasResourceRole`. See the [Spring Security page](https://www.baeldung.com/spring-security-check-user-role) for more examples.
+#### Modify Security Checks for Kubernetes
 
-### Local versus Kubernetes Testing
-This tutorial also makes use of the `internal` client configured in the microservice via the application.yml with roles manually created and assigned in Keycloak. In Kubernetes, Entando will automatically create client roles per the bundle plugin definition (see the plugin definition [here](../../../docs/curate/ecr-bundle-details.md) for more information). Those roles will be created for the client specific to the microservice itself, e.g. `<docker username>-conference-server`. This client name will be injected as an environment variable into the plugin container itself so the annotations noted above will work both in local and Kubernetes environments.
+The MFE authorization checks in this tutorial explicitly note the client ID,  e.g. `internal`. The following options modify the checks to work in Kubernetes:
 
-The MFE authorization checks in the tutorial explicitly note the client id,  e.g. `internal`, which won't work in Kubernetes. There are a couple options here:
-1) Change the application.yml clientId under `security.oauth2.client.registration.oidc` to match the Kubernetes clientId. That's the most secure and allows the MFE checks to work the same in both local and Kubernetes environments. However, you not be be able to use the same clientId depending on how the microservice is deployed
-2) An alternative is to broaden the MFE authorization check to look for a named role on any client. This could result in overlap with other clients but with appropriately named roles (e.g. prefixed by feature, e.g. `conference-admin`) this could be the most flexible option. This can be provided via a helper function, e.g. in `api/helpers.js`:
+1) Change the `application.yml` clientId under `security.oauth2.client.registration.oidc` to match the Kubernetes clientId. 
+
+   This is the most secure option and allows the MFE checks to work identically in both local and Kubernetes environments. However, you may not be be able to use the same clientId, depending on how the microservice is deployed.
+
+2) Broaden the MFE authorization check to look for a named role on any client. 
+
+   This could result in overlap with other clients, but with appropriately named roles (e.g. prefixed by feature, e.g. `conference-admin`) this is the most flexible option. It can be achieved via a helper function, e.g. in `api/helpers.js`, and results in a simpler role check:
```

Comment:
This could result in overlap with other clients, but with appropriately named roles (e.g. prefixed by feature, e.g. `conference-admin`) this is the most flexible option. >> the interruption of the parenthesis and confusion with two e.g.s makes this sentence hard to read and easy to loose the thread

Reply:
agreed the two "e.g."s are a bit awkward but think they make for effective phrasing to convey meaning as concisely as possible. can move flexibility before roles to not break up sentence.

Reply:
@nshaw to edit per jinah's suggestion:
(e.g. prefixed by feature, e.g. `conference-admin`) -> e.g. the role `conference-admin` on the client `prefixed by feature` ??